R/pkg/R/window.R

```diff
@@ -0,0 +1,88 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# window.R - Utility functions for defining window in DataFrames
+
+#' Window.partitionBy
+#'
+#' Creates a WindowSpec with the partitioning defined.
+#'
+#' @rdname Window.partitionBy
+#' @export
+#' @examples
+#' \dontrun{
+#'   Window.partitionBy("col1", "col2")
+#'   Window.partitionBy(df$col1, df$col2)
+#' }
+setMethod("Window.partitionBy",
```

Comment:
Naming in Scala: Window.partitionBy - https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowSuite.scala#L30
Window.orderBy - https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowSuite.scala#L41

dplyr supports window functions, see https://cran.r-project.org/web/packages/dplyr/vignettes/window-functions.html

but here window.* names are for factory methods in Scala, not window functions themselves. Refer to https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/expressions/Window.scala

I am ok with "window.*".